compiler/ilgen/MethodBuilder.cpp

```diff
@@ -139,7 +139,7 @@ MethodBuilder::asMethodBuilder()
 void
 MethodBuilder::initMaps()
    {
-   _parameterSlot = new (PERSISTENT_NEW) TR_HashTabString(typeDictionary()->trMemory());
+   _parameterSlot = std::map<const char *, int32_t>();
```

Comment:
I only refactored these initalizations to the constructor initializer list. Does this affect the memory allocation or lifetime of these objects in any way? Also, can you please explain what you mean by "inline" objects here?